packages/suite/src/hooks/wallet/useStakeEthForm.ts

```diff
@@ -0,0 +1,332 @@
+    };
+
+    const [state, setState] = useState<ReturnType<typeof useStakeEthState>>(undefined);
+
+    const { saveDraft, getDraft, removeDraft } = useFormDraft<StakeEthFormState>('stake-eth');
+    const draft = getDraft(account.key);
+    const isDraft = !!draft;
+
+    const { defaultValues } = useStakeEthFormDefaultValues(state?.formValues?.outputs[0].address);
+
+    // throttle initial state calculation
+    const initState = useStakeEthState(selectedAccount, fees, !!state, defaultValues);
+
+    useEffect(() => {
+        const setStateAsync = async (
+            initState: NonNullable<ReturnType<typeof useStakeEthState>>,
+        ) => {
+            const address = await getComposeAddressPlaceholder(account, network, device, accounts);
+            if (initState.formValues && address) {
+                initState.formValues.outputs[0].address = address;
+
+                setState(initState);
+            }
+        };
+
+        if (!state && initState) {
+            setStateAsync(initState);
+        }
+    }, [state, initState, account, network, device, accounts]);
+
+    const methods = useForm<StakeEthFormState>({
+        mode: 'onChange',
+        defaultValues: isDraft ? draft : defaultValues,
+    });
+
+    const { register, control, formState, setValue, reset, clearErrors, getValues, setError } =
+        methods;
+
+    const values = useWatch<StakeEthFormState>({ control });
+
+    useEffect(() => {
+        if (!isChanged(defaultValues, values)) {
+            removeDraft(account.key);
+        }
+    }, [defaultValues, values, removeDraft, account.key]);
+
+    // react-hook-form auto register custom form fields (without HTMLElement)
+    useEffect(() => {
+        register('outputs');
+        register('setMaxOutputId');
+    }, [register]);
+
+    // react-hook-form reset, set default values
+    useEffect(() => {
+        if (!isDraft && defaultValues) {
+            reset(defaultValues);
+        }
+    }, [reset, isDraft, defaultValues]);
+
+    const {
+        isLoading: isComposing,
+        composeRequest,
+        composedLevels,
+    } = useCompose({
+        ...methods,
+        state,
+    });
+
+    useDebounce(
+        () => {
+            if (
+                formState.isDirty &&
+                !formState.isValidating &&
+                Object.keys(formState.errors).length === 0 &&
+                !isComposing
+            ) {
+                saveDraft(selectedAccount.account.key, values as StakeEthFormState);
+            }
+        },
+        200,
+        [
+            saveDraft,
+            selectedAccount.account.key,
+            values,
+            formState.errors,
+            formState.isDirty,
+            formState.isValidating,
+            isComposing,
+        ],
+    );
+
+    const [isAmountForWithdrawalWarningShown, setIsAmountForWithdrawalWarningShown] =
+        useState(false);
+
+    const onCryptoAmountChange = useCallback(
```

Comment:
We extended the issue by 

```
Given I have a balance of 1 ETH
When I input more than '0.97' minus current ETH fee into the ETH field
Then I should see a message, 'We recommend you to leave 0.03 ETH so you will be able to pay for withdrawal fees"
```

Reply:
https://github.com/everstake/trezor-suite/pull/3/commits/1d1721ce124e600239df1fd8da961e4950bf2d3b